### ISSUE TITLE
IMemoryObject.TryGetString doesn't have correct nullable annotation

### ISSUE THREAD

**Issue by nevercast** [thomasfn/ScreepsDotNet#11]
Its my understanding that `IMemoryObject.TryGetString` should have `[MaybeNullWhen(false)]` annotation, the same as TryGetObject.

My IDE flags the out value as non-null, which I am confident is not always the case. 

https://github.com/thomasfn/ScreepsDotNet/blob/main/ScreepsDotNet.API/API/World/IMemoryObject.cs#L48

Have "patched" this locally in my bot with an Extension Method that does have a different signature but suffices.

```cs
  [return: NotNullIfNotNull("defaultValue")]
  public static string? TryGetString(this IMemoryObject memory, string key, string? defaultValue = null)
  {
    return memory.TryGetString(key, out var value) ? value : defaultValue;
  }
```

**Comment by thomasfn** [comment]
Good spot, I'll fix this.

**Comment by thomasfn** [comment]
Fixed in API 2.1.0
